knowledge-base/grid-kb-style-filtered-columns.md

```diff
@@ -0,0 +1,151 @@
+
+>caption Grid with styled filtered column.
+
+`````CSHTML
+@using Telerik.DataSource.Extensions
+@using Telerik.DataSource
+
+<style>
+    .highlighted-column {
+        background-color: rgba(255, 0, 0, 0.2); /* Change to your desired RGBA */
+        color: #191970;
+    }
+</style>
+
+<TelerikGrid TItem="@Employee" 
+             OnRead="@ReadItems"
+             FilterMode="@GridFilterMode.FilterRow"
+             Sortable="true" 
+             Pageable="true" 
+             OnStateChanged="@OnGridStateChanged">
+    <GridColumns>
+        <GridColumn Field="@nameof(Employee.ID)" Filterable="false" Title="ID" />
+        <GridColumn Field="@nameof(Employee.Name)" OnCellRender="@OnNameColumnCellRender" Title="Name" />
+        <GridColumn Field="@nameof(Employee.HireDate)" OnCellRender="@OnHireDateColumnCellRender" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Employee> SourceData { get; set; }
+    private string FilteredColumn { get; set; }
+
+    private void OnNameColumnCellRender(GridCellRenderEventArgs args)
+    {
+        if (FilteredColumn == nameof(Employee.Name))
+        {
+            args.Class = "highlighted-column"; // Apply the custom CSS class
+        }
+    }
+
+    private void OnHireDateColumnCellRender(GridCellRenderEventArgs args)
+    {
+        if (FilteredColumn == nameof(Employee.HireDate))
+        {
+            args.Class = "highlighted-column"; // Apply the custom CSS class
+        }
+    }
+
+    private async Task OnGridStateChanged(GridStateEventArgs<Employee> args)
+    {
+        if (args.PropertyName == "FilterDescriptors")
+        {
+            if (args.GridState.FilterDescriptors.Any())
+            {
+                var filterDescriptors = new List<IFilterDescriptor>(args.GridState.FilterDescriptors);
+                var filterDescriptor = filterDescriptors[0] as CompositeFilterDescriptor;
+                var filteredColumn = filterDescriptor.FilterDescriptors[0] as FilterDescriptor;
+
+                // Capture the filtered column's field name
+                FilteredColumn = filteredColumn.Member;
+            }
+            else
+            {
+                FilteredColumn = null; // Reset when no filters are applied
+            }
+
+            StateHasChanged(); // Re-render the grid
+        }
+    }
+
+    protected async Task ReadItems(GridReadEventArgs args)
+    {
+        // Simulate network delay
+        await Task.Delay(100);
+
+        var datasourceResult = SourceData.ToDataSourceResult(args.Request);
+
+        args.Data = datasourceResult.Data;
+        args.Total = datasourceResult.Total;
+    }
+
+    protected override void OnInitialized()
+    {
+        SourceData = GenerateData();
+    }
+
+    private List<Employee> GenerateData()
+    {
+        var result = new List<Employee>();
+        var rand = new Random();
+        for (int i = 0; i < 100; i++)
```

Comment:
Set ID values from 1. Using the default value of a type as a valid ID value can backfire.